File: snippets/python_snippets/getting_started.py
import breez_sdk

import logging

sdk_services = breez_sdk.BlockingBreezServices
# ANCHOR: init-sdk
class SDKListener(breez_sdk.EventListener):
    def on_event(self, event):
        logging.info(event)

def getting_started():
    
    seed = breez_sdk.mnemonic_to_seed("<mnemonic words>")
    invite_code = "<invite code>"
    api_key = "<api key>"
    config = breez_sdk.default_config(
        breez_sdk.EnvironmentType.PRODUCTION, 
        api_key, 
        breez_sdk.NodeConfig.GREENLIGHT(breez_sdk.GreenlightNodeConfig(None, invite_code)))

    # Customize the config object according to your needs
    config.working_dir = "path to an existing directory"

    try:
        # Connect to the Breez SDK make it ready for use
        sdk_services = breez_sdk.connect(config, seed, SDKListener())
        
        return sdk_services
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: init-sdk

def getting_started_node_info():
     
    try: 
         # ANCHOR: fetch-balance
        node_info = node_info()
        ln_balance = node_info.channels_balance_msat
        onchain_balance = node_info.onchain_balance_msat
        # ANCHOR_END: fetch-balance
    except Exception as error:
        logging.error(error)
        raise

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/GettingStarted.kt
package com.example.kotlinmpplib

import breez_sdk.*
class GettingStarted {
    fun start() {
        // ANCHOR: init-sdk
        // SDK events listener
        class SDKListener : EventListener {
            override fun onEvent(e: BreezEvent) {
                // Log.v("SDKListener", "Received event $e")
            }
        }

        // Select your seed, invite code and eviroment
        val seed = mnemonicToSeed("<mnemonic words>")
        val inviteCode = "<invite code>"
        val apiKey = "<api key>"

        // Create the default config
        val greenlightNodeConfig = GreenlightNodeConfig(null, inviteCode)
        val nodeConfig = NodeConfig.Greenlight(greenlightNodeConfig)
        val config = defaultConfig(EnvironmentType.PRODUCTION, apiKey, nodeConfig)

        // Customize the config object according to your needs
        config.workingDir = "path to an existing directory"

        try {
            // Connect to the Breez SDK make it ready for use
            val connectRequest = ConnectRequest(config, seed)
            val sdk = connect(connectRequest, SDKListener())
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: init-sdk
    }

    fun startRestoreOnly(config: Config, seed: List<UByte>) {
        class SDKListener : EventListener {
            override fun onEvent(e: BreezEvent) {}
        }

        try {
            // ANCHOR: init-sdk-restore-only
            val connectRequest = ConnectRequest(config, seed, true)
            val sdk = connect(connectRequest, SDKListener())
            // ANCHOR_END: init-sdk-restore-only
        } catch (e: Exception) {
            // handle error
        }
    }

    fun fetchBalance(sdk: BlockingBreezServices) {
        // ANCHOR: fetch-balance
        try {
            val nodeInfo = sdk.nodeInfo()
            val lnBalance = nodeInfo?.channelsBalanceMsat
            val onchainBalance = nodeInfo?.onchainBalanceMsat
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: fetch-balance
    }

    fun logging() {
        // ANCHOR: logging
        class SDKLogStream : LogStream {
            override fun log(l: LogEntry) {
                // Log.v("SDKListener", "Received log [${l.level}]: ${l.line}")
            }
        }

        try {
            setLogStream(SDKLogStream())
        } catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: logging
    }

    // ANCHOR: disconnect
    fun disconnect(sdk: BlockingBreezServices)  {
        try {
            sdk.disconnect()
        } catch (e: Exception) {
            // handle error
        }
    }
    // ANCHOR_END: disconnect
}